File: String mirror[7-May-2023]/Solution.java
//  >> JAVA CODE <<

public static String stringMirror(String s) {
        String k = "" + s.charAt(0);
        for (int i = 1; i < s.length(); ++i) {
            if (s.charAt(0) == s.charAt(1)) {
                break;
            }
            if (s.charAt(i) <= s.charAt(i-1)) {
                k += s.charAt(i);
            } else {
                break;
            }
        }
        k += new StringBuilder(k).reverse().toString();
        return k;
    }

File: String mirror[7-May-2023]/Solution.py
#  >> PYTHON CODE <<

 def stringMirror(self, s : str) -> str:

        k=s[0]

        for i in range(1,len(s)):

            if s[0]==s[1]:

                break

            if s[i]<=s[i-1]:

                k+=s[i]

            else:

                break

        k+=k[::-1]

        return k

